lib/cocoapods/generator/xcconfig/xcconfig_helper.rb

```diff
@@ -196,6 +203,47 @@ def self.add_target_specific_settings(target, xcconfig)
           add_language_specific_settings(target, xcconfig)
         end
 
+        # Add the search paths for frameworks and libraries the given target
+        # depends on, so that it can be correctly built and linked.
+        #
+        # @param  [Target] target
+        #         The target.
+        #
+        # @param  [Array<PodTarget>] dependent_targets
+        #         The pod targets the given target depends on.
+        #
+        # @param  [Xcodeproj::Config] xcconfig
+        #         The xcconfig to edit.
+        #
+        def self.add_settings_for_dependent_targets(target, dependent_targets, xcconfig)
+          dependent_targets = dependent_targets.select(&:should_build?)
+          has_configuration_build_dir = target.respond_to?(:configuration_build_dir)
+          if has_configuration_build_dir
+            build_dir_var = "$#{SHARED_BUILD_DIR_VARIABLE}"
+            build_settings = {
+              'CONFIGURATION_BUILD_DIR' => target.configuration_build_dir(build_dir_var),
+              SHARED_BUILD_DIR_VARIABLE => '$(BUILD_DIR)/$(CONFIGURATION)$(EFFECTIVE_PLATFORM_NAME)',
+            }
+          else
+            build_dir_var = '$CONFIGURATION_BUILD_DIR'
+            build_settings = {}
+          end
+          unless dependent_targets.empty?
+            framework_search_paths = []
+            library_search_paths = []
+            dependent_targets.each do |dependent_target|
+              if dependent_target.requires_frameworks?
+                framework_search_paths << dependent_target.configuration_build_dir(build_dir_var)
+              else
+                library_search_paths << dependent_target.configuration_build_dir(build_dir_var)
+              end
+            end
+            build_settings['FRAMEWORK_SEARCH_PATHS'] = '$(inherited) ' + XCConfigHelper.quote(framework_search_paths.uniq) unless framework_search_paths.empty?
```

Comment:
Just an attempt into minimizing the footprint of the xcconfig. I'm happy to waive that.